Libraries/Text/RCTTextView.m

```diff
@@ -162,6 +162,23 @@ - (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range r
   }
 }
 
+- (void)setSelectionRange:(NSDictionary *)selectionRange
+{
+  NSInteger eventLag = _nativeEventCount - _mostRecentEventCount;
+  if (eventLag == 0) {
+    NSInteger selectionStart = [RCTConvert NSInteger:selectionRange[@"start"]];
+    NSInteger selectionEnd = [RCTConvert NSInteger:selectionRange[@"end"]];
+    UITextPosition *start = [_textView positionFromPosition:[_textView beginningOfDocument]
+                                                     offset:selectionStart];
+    UITextPosition *end = [_textView positionFromPosition:[_textView beginningOfDocument]
+                                                   offset:selectionEnd];
+    UITextRange *selection = [_textView textRangeFromPosition:start toPosition:end];
+    _textView.selectedTextRange = selection;
+  } else if (eventLag > RCTTextUpdateLagWarningThreshold) {
+    RCTLogWarn(@"Native TextInput(%@) is %zd events ahead of JS - try to make your JS faster.", self.text, eventLag);
```

Comment:
This isn't a TextInput ... may want to share this code so there's less copy & pasting going on.